app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -155,16 +156,32 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        this.fromManager = this.getIntent().
-                getBooleanExtra(AppManagerActivity.KEY_LAUNCH_FROM_MANAGER, false);
+        fromManager = getIntent().getBooleanExtra(AppManagerActivity.KEY_LAUNCH_FROM_MANAGER, false);
 
-        if (checkForMultipleAppsViolation()) {

```

Comment:
Would you mind actually leaving this logic the way it was? I know you don't like this pattern in general, but in scenarios like this where it's really important that no code be called after this line if it returns `true`, it feels like it will be much easier for someone in the future to accidentally add code after this `if` statement that messes that up.